SPEC.md

```diff
@@ -47,12 +47,14 @@ https://docs.google.com/a/coreos.com/document/d/1CTAL4gwqRofjxyp4tTkbgHtAwb2YCcP
 - Upon completion of the container lifecycle, the runtime must execute the plugins in reverse order (relative to the order in which they were executed to add the container) to disconnect the container from the networks.
 - The container runtime must not invoke parallel operations for the same container, but is allowed to invoke parallel operations for different containers.
 - The container runtime must order ADD and DEL operations for a container, such that ADD is always followed by a corresponding DEL. DEL may be followed by additional DELs, however, and plugins should handle multiple DELs permissively (i.e. plugin DEL should be idempotent).
+- A container is uniquely identified by a ContainerID. Plugins that store state should do so using a primary key of `(network name, container id)`. 
+- A runtime must not call ADD twice (without a corresponding DEL) for the same `(network name, container id)`. In other words, a given container ID must be added to a specific network exactly once.
```

Comment:
Interesting.  I can't think of a good use case for a container to have two interfaces on the same network.  But nevertheless, the spec currently allows it.  I wonder if this deserves a version bump.

Reply:
We can bundle it in with GET for v0.4.0

Reply:
Hmm, what about bond-type use-cases.  Assuming there are two physical interfaces placed into the container and then bonded together.  This could be done by a "master" plugin that delegates to two "physical device" plugins, and in that case the delegated plugin would get the same name/id tuple twice (but perhaps different config).

Reply:
Given that the container ends up with a single interface with a single address scope (i.e. the bonded ifs don't have addresses), it seems that bonding would be a single CNI configuration.